### PR TITLE
feat(ecs-patterns): allow customizing logdriver

### PR DESCRIPTION
**Pull request by piradeepk**
Provide the ability to pass logdriver as a property of all ecs-patterns constructs and update all constructs to accept either a vpc or cluster.

----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pkandasamy91** [comment]
@kohidave please review

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts [hunk 1]
```diff
@@ -143,6 +143,13 @@ export interface LoadBalancedServiceBaseProps {
    * @default CloudFormation-generated name
    */
   readonly serviceName?: string;
+
+  /**
+   * The LogDriver to use for logging.
+   *
+   * @default AwsLogDriver if enableLogging is true
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts]: Use “-“ for the default value if it’s not a real value

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts [hunk 2]
```diff
@@ -116,6 +135,11 @@ export abstract class QueueProcessingServiceBase extends Construct {
    */
   public readonly desiredCount: number;
 
+  /**
+   * The minimum number of tasks to run
+   */
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts]: Docstrings is wrong

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts [hunk 3]
```diff
@@ -136,6 +160,11 @@ export abstract class QueueProcessingServiceBase extends Construct {
   constructor(scope: Construct, id: string, props: QueueProcessingServiceBaseProps) {
     super(scope, id);
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts]: Feels like we can probably reuse this logic across patterns (via common utility function)

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts (hunk 1 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts]: Updated

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts (hunk 2 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts]: Addressed.

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts (hunk 3 above)
Comment (pkandasamy91) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts]: Potentially, I'll leave this PR as is, and create an issue to discuss this, there's a lot of common functionality that's separated today.

**Comment by eladb** [comment]
See my title update

**Comment by pkandasamy91** [comment]
Rebased with master

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts [hunk 4]
```diff
@@ -178,7 +185,7 @@ export abstract class LoadBalancedServiceBase extends cdk.Construct {
 
     // Create log driver if logging is enabled
     const enableLogging = props.enableLogging !== undefined ? props.enableLogging : true;
-    this.logDriver = enableLogging ? this.createAWSLogDriver(this.node.id) : undefined;
+    this.logDriver = props.logDriver !== undefined ? props.logDriver : enableLogging ? this.createAWSLogDriver(this.node.id) : undefined;
```
Comment (kohidave) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts]: nit - Maybe we can break this up into multiple lines so it's a little easier to follow.

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts [hunk 5]
```diff
@@ -145,7 +174,7 @@ export abstract class QueueProcessingServiceBase extends Construct {
 
     // Create log driver if logging is enabled
     const enableLogging = props.enableLogging !== undefined ? props.enableLogging : true;
-    this.logDriver = enableLogging ? this.createAWSLogDriver(this.node.id) : undefined;
+    this.logDriver = props.logDriver !== undefined ? props.logDriver : enableLogging ? this.createAWSLogDriver(this.node.id) : undefined;
```
Comment (kohidave) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts]: Same comment as above - can we break this up into a few lines?

**Pull request merged**